Index: src/store/bookSlice.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/store/bookSlice.ts b/src/store/bookSlice.ts
--- a/src/store/bookSlice.ts	
+++ b/src/store/bookSlice.ts	
@@ -1,22 +1,32 @@
 import {createSlice} from '@reduxjs/toolkit'
 import type {PayloadAction} from '@reduxjs/toolkit'
-import {BookList} from '../types/index'
+import {BookList, QueryParams, Books} from '../types/index'
 
 const initialState: BookList = {
     books: [],
     totalBooks: 0,
+    getSearch: 'Mark Twain',
+    selectedTheme: 'all',
+    sortedResult: 'relevance',
+    startIndex: 0,
 }
 
 export const bookSlice  = createSlice({
     name: 'booksAPI',
     initialState,
     reducers: {
-        addBooksToState: (state, action: PayloadAction<BookList>) => {
+        addBooksToState: (state, action: PayloadAction<Books>) => {
             state.books = [...action.payload.books];
             state.totalBooks = action.payload.totalBooks;
         },
+        setQueryParams: (state, action: PayloadAction<QueryParams>) => {
+            state.getSearch = action.payload.getSearch;
+            state.selectedTheme = action.payload.selectedTheme;
+            state.sortedResult = action.payload.sortedResult;
+            state.startIndex = action.payload.startIndex;
+        },
     },
 })
 
-export const {addBooksToState} = bookSlice.actions
+export const {addBooksToState, setQueryParams} = bookSlice.actions
 export const booksReducer =  bookSlice.reducer
Index: src/types/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/types/index.ts b/src/types/index.ts
--- a/src/types/index.ts	
+++ b/src/types/index.ts	
@@ -3,7 +3,18 @@
     value: string;
 }
 
-export interface BookList {
+export interface BookList extends QueryParams, Books{
+
+}
+
+export interface QueryParams {
+    getSearch: string;
+    selectedTheme: string;
+    sortedResult: string;
+    startIndex: number;
+}
+
+export interface Books {
     books: Book[];
     totalBooks: number;
 }
@@ -22,4 +33,4 @@
 export interface ImageLinks {
     thumbnail: string;
     smallThumbnail: string;
-}
+}
\ No newline at end of file
